app/models/barcode/format_handlers.rb

```diff
@@ -503,4 +503,15 @@ class UkBiocentreV7 < BaseRegExBarcode
   class EastLondonGenesAndHealth < BaseRegExBarcode
     self.format = /\A(?<prefix>S2)-046-(?<number>\d+)\z/
   end
+
+  # Support for East London Genes and Health
+  # As part of the Cardinal pipeline, as above
+  # Expected formats:
+  # S2-nnn-nnnnnn
+  # where n is a digit
+  # This is needed in addition to the above because the middle section represents the study
+  # in ELGH's system, and some participants come in under a different primary study.
+  class EastLondonGenesAndHealthV2 < BaseRegExBarcode
+    self.format = /\A(?<prefix>S2)-\d+-(?<number>\d+)\z/
```

Comment:
that doesn't restrict it to 3 digits in the middle, is that ok?

Reply:
If you want it to, just put `{3}` after `\d` instead of `+`

Reply:
I thought I'd leave it more flexible, as they didn't actually say it would always be 3 digits

Reply:
Actually they kind of did. I don't know. I'll leave it open.

Reply:
There is a part of the barcode that is not classified as <prefix> or <number>, why it couldnt go as prefix?

Reply:
Hi Eduardo, sorry I think I merged this just before you commented, or before I saw your comment. I didn't quite know where to put the middle section of the barcode. The other barcode formats seemed to have a static prefix, and this middle bit is dynamic. I don't know if that matters!